rio_tiler/io/stac.py

```diff
@@ -148,7 +156,7 @@ class STACReader(MultiBaseReader):
     """
 
     filepath: str = attr.ib()
-    item: Dict = attr.ib(default=None)
+    item: pystac.Item = attr.ib(default=None, converter=to_stac_item)
```

Comment:
🤔 I think `converter` will always be run. we got lucky because `None` will be returned directly (not caught by `if isinstance...`) I'm fine with this because honestly I have no better way to do it 

Example:
```python
@attr.s
class a():
    b : int = attr.ib(default=None, converter=int)

a()
> TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'
```